units/ftdc_rollups.go

```diff
@@ -0,0 +1,129 @@
+package units
+
+import (
+	"context"
+
+	"github.com/evergreen-ci/cedar"
+	"github.com/evergreen-ci/cedar/model"
+	"github.com/evergreen-ci/cedar/perf"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/ftdc"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+const (
+	ftdcRollupsJobName = "ftdc-rollups"
+)
+
+type ftdcRollups struct {
+	PerfID       string              `bson:"perf_id" json:"perf_id" yaml:"perf_id"`
+	ArtifactInfo *model.ArtifactInfo `bson:"artifact" json:"artifact" yaml:"artifact"`
+
+	*job.Base `bson:"metadata" json:"metadata" yaml:"metadata"`
+}
+
+func init() {
+	registry.AddJobType(ftdcRollupsJobName, func() amboy.Job {
+		fr := &ftdcRollups{}
+		fr.setup()
+		fr.SetDependency(dependency.NewAlways())
+
+		return fr
+	})
+}
+
+func (fr *ftdcRollups) setup() {
+	if fr.Base == nil {
+		fr.Base = &job.Base{
+			JobType: amboy.JobType{
+				Name:    ftdcRollupsJobName,
+				Version: 1,
+			},
+		}
+	}
+}
+
+func (fr *ftdcRollups) Validate() error {
+	fr.setup()
+	fr.SetDependency(dependency.NewAlways())
```

Comment:
I think you could drop this by just having a seperate constructor and follow the pattern that all of the evergreen jobs have... (the duplication and difference between the environment that you test in and would use in actual execution (because of the factory) case coul,d lead to some interesting bugs.